extras/parse_travis_log.py

```diff
@@ -191,13 +191,41 @@ def print_static_analysis(d):
                 print(INDENT + ' | ' + 2 * INDENT +
                       ' - ' + k3 + ' : ' + str(v3))
 
+def count_warnings_errors(f):
+    """
+    Counts compiler warnings and errors. Stops when reading '+make install'.
+    """
+    warn = {}
+    error = {}
+    while True:
+        line = f.readline()
+        if not line:
+            return warn, error, line
+
+        if line.strip() == '+make install':
+            return warn, error, line
+
+        # /home/travis/build/tammoippen/nest-simulator/build/libnestutil/config.h:65:0: warning: "HAVE_DIRENT_H" redefined [enabled by default]
+        if ': warning:' in line:
+            file = line.split(':')[0]
+            if file not in warn:
+                warn[file] = 0
+            warn[file] += 1
+
+        if ': error:' in line:
```

Comment:
I am not really fond of a string constant here, as compilers may change. But gcc 4 and 5, clang, Intel and Portland Group compilers all include this string in their error messages, so I guess it is a rather safe bet. `xlC` will not work, but I think that is not a problem for checking on Travis.